pkg/agent/handler/accessrequest.go

```diff
@@ -57,7 +58,7 @@ func (h *accessRequestHandler) Handle(action proto.Event_Type, meta *proto.Event
 		return nil
 	}
 
-	if ar.Status.Level != statusPending {
+	if ar.Status.Level != statusPending && !(ar.Status.Level == statusSuccess && ar.Metadata.State == v1.ResourceDeleting) {
```

Comment:
I was thinking about adding a func to figure out when to process based on status. The addition of the state checks makes me think it's time for that. I can do it though once this is merged if you want.